### ISSUE TITLE
Flux Integration

### ISSUE THREAD

**Issue by MikeInnes** [dfdx/Yota.jl#26]
I'm curious if you'd be interested in making Yota compatible with Flux layers and optimisers; then Yota could be used in place of Tracker for models without control flow.

Zygote does this by inserting calls to [unwrap](https://github.com/FluxML/Zygote.jl/blob/020c26b6828ba9ef80da40436ad442196e177861/src/flux.jl#L5) which strip away Flux tracking (this of course won't be necessary when we get rid of Tracker).

**Comment by dfdx** [comment]
Yes, I thought about it, but I don't see a straightforward way to do it just yet - unwrapping tracked data helps in finding gradients, but how do you use them after that? 

For example, let's take [`_update_params!(opt, xs)`](https://github.com/FluxML/Flux.jl/blob/master/src/optimise/train.jl#L9-L15) - it expects each `x` to  have `.data` and `.grad` properties, e.g. be a `Tracked*` type. Zygote by itself doesn't use tracked data, do you just record found gradients back to original tracked variables? If so, do you have a pointer to the relevant piece of code?

**Comment by MikeInnes** [comment]
I spent some time refactoring this stuff for exactly this reason. We now have [this version of update](https://github.com/FluxML/Flux.jl/blob/3a4c6274fadf1b94468da455a06f070b2cda6a64/src/optimise/train.jl#L9) which only requires a param -> grad mapping. It should be pretty easy to get that out of Yota.

The `Param` API is somewhat transitional, it represents a compromise between what Flux and Yota/Zygote can expose, but the idea is to eventually get rid of it in favour of [more functional optimisers](https://github.com/FluxML/Flux.jl/issues/637). Once that happens it should be possible to use Yota's native API with Flux.
